### PR TITLE
docs(esl-media-query): improve documentation

### LINKED ISSUE #1865: (📖esl-media-query): improve `esl-media-query` documentation (site and README)
There are some API and usage details are missed in the public Readme
Especially Event types

### PR DESCRIPTION
**Pull request by fshovchko**
Closes #1865

### CODE REVIEW

File: src/modules/esl-media-query/README.md [hunk 1]
```diff
@@ -126,22 +79,96 @@ class Test {
...
+  - `ignoreBotsDpr` - marker to enable DPR ignoring for PageSpeed Bots
+  - `toDPI` - converts to DPI (Dots Per Inch) based on a given number value
+  - `process` - creates a string representation of the DPI value, in the format of (e.g. `(min-resolution: {value}dpi)`)
+
+---
+
+<a name="shortcuts"></a>
+### ESLEnvShortcuts
+
+ESLEnvShortcuts is a simple registry for a static shortcuts to describe environment related conditions
```
Comment (NastaLeo) [inline src/modules/esl-media-query/README.md:120]:
```suggestion
`ESLEnvShortcuts` is a simple registry for a static shortcuts to describe environment related conditions.
```

File: src/modules/esl-media-query/README.md [hunk 2]
```diff
@@ -126,22 +79,96 @@ class Test {
...
+
+An additional shortcuts can be added to the registry through `ESLEnvShortcuts.add` method
+ESLEnvShortcuts allows adding boolean result that will be converted to `all` / `not all` query conditions or setup result as native MediaQuery string.
```
Comment (NastaLeo) [inline src/modules/esl-media-query/README.md:123]:
```suggestion
ESLEnvShortcuts allows adding boolean result that will be converted to `all` / `not all` query condition or setup result as native MediaQuery string. 
```

File: src/modules/esl-media-query/README.md [hunk 3]
```diff
@@ -126,22 +79,96 @@ class Test {
...
 Pair of `ESLMediaQuery` and value (payload) associated with a query condition. 
-`ESLMediaRule` is used as an item for [ESLMediaRuleList](#eslmediarulelist).  
+`ESLMediaRule` is used as an item for [ESLMediaRuleList](#media-rule-list).  
 `ESLMediaRule` can be parsed from `<ESL Media Query> => <value>` syntax string, 
 e.g. `@XS => 1` (`1` is the payload) or `@+LG and @DESKTOP => desktop` (`'desktop'` is the payload).
 
+#### ESLMediaRule API
+- `ESLMediaRule.prototype.create(payload, query, parser)` - creates an `ESLMediaRule` instance from a payload string, media query, and a payload parser function. If the payload parsing result is undefined, the rule will also be undefined
+- `ESLMediaRule.prototype.parse` - parses media rule defined with syntax, where query and value separated by `=>`
+- `ESLMediaRule.prototype.all(payload)` - creates an always active `ESLMediaRule` with passed value
```
Comment (NastaLeo) [inline src/modules/esl-media-query/README.md:154]: mb will be better 'always' place at the start of phrase

File: src/modules/esl-media-query/README.md [hunk 4]
```diff
@@ -126,22 +79,96 @@ class Test {
...
 ```
 
+The incoming JavaScript event will be an instance of [ESLMediaChangeEvent](#media-change-event).
+
+---
+
+<a name="breakpoints"></a>
+### ESLScreenBreakpoints
+
+ESLScreenBreakpoints is a custom screen breakpoints registry. It allows defining a named screen width range.
```
Comment (NastaLeo) [inline src/modules/esl-media-query/README.md:89]:
```suggestion
`ESLScreenBreakpoints` is a custom screen breakpoints registry. It allows defining a named screen width range.
```

File: src/modules/esl-media-query/README.md [hunk 5]
```diff
@@ -204,3 +231,26 @@ Event Target methods:
...
 `ESLMediaRuleList.prototype.removeEventListener('change', cb: EventListener)` - unsubscribes from `ESLMediaRuleList` object value changes 
 
+The incoming JavaScript event will be an instance of [ESLMediaRuleListEvent](#media-rule-event).
+
+---
+### ESLMediaQuery events
+<a name="media-rule-event"></a>
+#### ESLMediaRuleListEvent
+
+`ESLMediaRuleListEvent` is a custom event dispatched by `ESLMediaRuleList` instances
```
Comment (NastaLeo) [inline src/modules/esl-media-query/README.md:241]:
```suggestion
`ESLMediaRuleListEvent` is a custom event dispatched by `ESLMediaRuleList` instances.
```

File: src/modules/esl-media-query/README.md [hunk 6]
```diff
@@ -204,3 +231,26 @@ Event Target methods:
...
+
+`ESLMediaRuleListEvent` properties:
+  - `current` - current value of target `ESLMediaRuleList` instances
```
Comment (NastaLeo) [inline src/modules/esl-media-query/README.md:244]:
```suggestion
  - `current` - current value of target `ESLMediaRuleList` instance
```

File: src/modules/esl-media-query/README.md [hunk 7]
```diff
@@ -204,3 +231,26 @@ Event Target methods:
...
+  - `previous` - previous value of target `ESLMediaRuleList` instances
```
Comment (NastaLeo) [inline src/modules/esl-media-query/README.md:245]:
```suggestion
  - `previous` - previous value of target `ESLMediaRuleList` instance
```

File: src/modules/esl-media-query/README.md [hunk 8]
```diff
@@ -204,3 +231,26 @@ Event Target methods:
...
+  - `target` - target `ESLMediaRuleList` instance
+
+<a name="media-change-event"></a>
+#### ESLMediaChangeEvent
+
+`ESLMediaChangeEvent` is a custom event dispatched by `ESLMediaQuery` instances
```
Comment (NastaLeo) [inline src/modules/esl-media-query/README.md:251]:
```suggestion
`ESLMediaChangeEvent` is a custom event dispatched by `ESLMediaQuery` instances.
```

**Review by NastaLeo — APPROVED** [review]

File: src/modules/esl-media-query/README.md (hunk 3 above)
Comment (fshovchko) [reply src/modules/esl-media-query/README.md:154]: sounds okay both ways for me

**Review by dshovchko — APPROVED** [review]

**Review by ala-n — APPROVED** [review]

**Pull request merged**